src/provider/index.ts

```diff
@@ -0,0 +1,5 @@
+import * as slack from './slack';
+
+export * as slack from './slack';
+
+export type INative = slack.INative;
```

Comment:
How is this supposed to scale to other services?

Reply:
So Slack is blessed by claiming the generic-sounding `INative`? What names would be exported from here for other providers? And how are they consumed? I see it surfaces in identity.ts as 

```
interface IEveryAuthCredential {
  native: provider.INative /** The raw identity from a service. */;
  accessToken: string /** The OAuth2 accessToken used to authenticate a request. */;
}
```

Does this surface all the way to the public OM in a way that is typed and helps with completion?

Reply:
No, it becomes:
```
export type INative = slack.INative | salesforce.INative | hubspot.INative; // etc
```

It's not super elegant because we're not doing any of the fancy auto-type-discovery stuff we do in the editor.